astropy/units/tests/test_physical.py

```diff
@@ -1,60 +1,430 @@
+    (u.C * u.m, "electrical dipole moment"),
+    (u.A / u.m ** 2, "electrical current density"),
+    (u.V / u.m, "electrical field strength"),
+    (u.C / u.m ** 2, "electrical flux density"),
+    (u.C / u.m ** 3, "electrical charge density"),
+    (u.F / u.m, "permittivity"),
+    (u.Wb, "magnetic flux"),
+    (u.T, "magnetic flux density"),
+    (u.A / u.m, "magnetic field strength"),
+    (u.H / u.m, "electromagnetic field strength"),
+    (u.H, "inductance"),
+    (u.cd, "luminous intensity"),
+    (u.lm, "luminous flux"),
+    (u.lx, "luminous emittance/illuminance"),
+    (u.W / u.sr, "radiant intensity"),
+    (u.cd / u.m ** 2, "luminance"),
+    (u.astrophys.Jy, "spectral flux density"),
+    (u.astrophys.R, "photon flux"),
+    (u.misc.bit, "data quantity"),
+    (u.misc.bit / u.s, "bandwidth"),
+    (u.cgs.Franklin, "electrical charge (ESU)"),
+    (u.cgs.statampere, "electrical current (ESU)"),
+    (u.cgs.Biot, "electrical current (EMU)"),
+    (u.cgs.abcoulomb, "electrical charge (EMU)"),
+    (u.imperial.btu / (u.s * u.m * u.imperial.deg_F), "thermal conductivity"),
+    (u.imperial.cal / u.deg_C, "heat capacity"),
+    (u.imperial.cal / u.deg_C / u.g, "specific heat capacity"),
+    (u.J * u.m ** -2 * u.s ** -1, "energy flux"),
+    (u.W / u.m ** 2, "energy flux"),
+    (u.m ** 3 / u.mol, "molar volume"),
+    (u.m / u.S, "electrical resistivity"),
+    (u.S / u.m, "electrical conductivity"),
+    (u.A * u.m ** 2, "magnetic moment"),
+    (u.J / u.T, "magnetic moment"),
+    (u.yr ** -1 * u.Mpc ** -3, "volumetric rate"),
+    (u.m / u.s ** 3, "jerk"),
+    (u.m / u.s ** 4, "snap"),
+    (u.m / u.s ** 5, "crackle"),
+    (u.m / u.s ** 6, "pop"),
+    (u.deg_C / u.m, "temperature gradient"),
+    (u.imperial.deg_F / u.m, "temperature gradient"),
+    (u.imperial.deg_R / u.imperial.ft, "temperature gradient"),
+    (u.imperial.Calorie / u.g, "specific energy"),
+    (u.mol / u.L / u.s, "reaction rate"),
+    (u.imperial.lbf * u.imperial.ft * u.s ** 2, "moment of inertia"),
+    (u.mol / u.s, "catalytic activity"),
+    (u.imperial.kcal / u.deg_C / u.mol, "molar heat capacity"),
+    (u.mol / u.kg, "molality"),
+    (u.imperial.inch * u.hr, "absement"),
+    (u.imperial.ft ** 3 / u.s, "volumetric flow rate"),
+    (u.Hz / u.s, "frequency drift"),
+    (u.Pa ** -1, "compressibility"),
+    (u.count, "number of counts"),
+    (u.electron, "number of electrons"),
+    (u.pix, "number of elements on 2D regular grid"),
+    (u.vox, "number of elements on 3D regular grid"),
+]
 
 
-def test_power():
-    assert (u.cm ** 3).physical_type == 'volume'
+@pytest.mark.parametrize("unit, physical_type", unit_physical_type_pairs)
+def test_physical_types(unit, physical_type):
+    """
+    Test that the `physical_type` attribute of `u.Unit` objects provides
+    the expected physical type for various units.
 
+    Many of these tests are used to test backwards compatibility.
+    """
+    if unit.physical_type != physical_type:
```

Comment:
Why not just `assert unit.physical_type == physical_type`? Pytest will then list essentially the failure message added here, no?